exercises/simple-linked-list/example.js

```diff
@@ -1,111 +1,52 @@
-export class ElementValueRequiredException extends Error {}
-export class ElementNextNotInstanceException extends Error {}
-
 export class Element {
-  constructor(value, next) {
-    if (value === undefined) {
-      throw new ElementValueRequiredException();
-    }
-
-    if (next !== undefined && !(next instanceof Element)) {
-      throw new ElementNextNotInstanceException();
-    }
-
+  constructor(value) {
     this.value = value;
-    this.next = next;
+    this.next = null;
   }
 }
 
 export class List {
-  static fromArray(array) {
-    const list = new List();
-    array.forEach(item => list.push(new Element(item)));
-
-    return list;
-  }
-
-  push(value) {
-    const newEl = (value instanceof Element)
-      ? value
-      : new Element(value);
-
-    if (!this.head) {
-      this.head = newEl;
-      return;
-    }
-
-    let lastEl = this.head;
-    while (lastEl.next) {
-      lastEl = lastEl.next;
+  constructor(arr) {
+    this.head = null;
+    if (arr) {
+      arr.forEach(el => this.add(new Element(el)));
     }
-
-    lastEl.next = newEl;
   }
 
-  unshift(value) {
-    const newEl = (value instanceof Element)
-      ? value
-      : new Element(value);
-
-    newEl.next = this.head;
-    this.head = newEl;
+  get length() {
+    return this.head ? this.countElements(1, this.head) : 0;
   }
 
-  shift() {
-    if (!this.head) {
-      return;
-    }
-
-    this.head = this.head.next;
+  add(el) {
+    const element = el;
+    element.next = this.head;
+    this.head = element;
   }
 
-  pop() {
-    if (!this.head) {
-      return;
-    }
-
-    let penultEl;
-    let lastEl = this.head;
-
-    while (lastEl.next) {
-      penultEl = lastEl;
-      lastEl = lastEl.next;
-    }
-
-    if (!penultEl) {
-      this.head = undefined;
-    } else {
-      penultEl.next = undefined;
+  countElements(c, element) {
+    let count = c;
+    while (element.next) {
+      count += 1;
+      return this.countElements(count, element.next);
```

Comment:
You can use either one of loop or recursion. Just like you did with `toArray`. A return condition before recursion call should work.

Reply:
Yes!  I think `reverse` can be recursive also.

I'm going to make those changes as soon as I have a few minutes.